grafana/grafana
# Loki: Improve handling of empty responses

helps with https://github.com/grafana/grafana/issues/52046

the loki http response parser calls `converter.ReadPrometheusStyleResult` (https://github.com/grafana/grafana/blob/main/pkg/util/converter/prom.go#L27), which can return a nil-pointer. the loki code did not handle this case. 

the linked github-issue describes such a situation.

this pull-request adds code to handle the nil-pointer case.

how to test:
- the new code is covered by a test, so it should be ok
- try to run some generic loki queries, to make sure this did not break something else


(NOTE:  this will not resolve the reported issue (why are we not getting back any data for the log-volume-query), but we should handle the nil-pointer-case.
